wcfsetup/install/files/lib/form/RegisterForm.class.php

```diff
@@ -164,6 +174,13 @@ public function validate() {
 		if (!$this->isExternalAuthentication && (!WCF::getSession()->getVar('registrationStartTime') || (TIME_NOW - WCF::getSession()->getVar('registrationStartTime')) < self::$minRegistrationTime)) {
 			throw new UserInputException('registrationStartTime', []);
 		}
+		
+		if (BLACKLIST_SFS_ENABLE) {
+			$this->blacklistMatches = BlacklistEntry::getMatches($this->username, $this->email, UserUtil::getIpAddress());
+			if (BLACKLIST_SFS_ACTION === 'block') {
+				throw new PermissionDeniedException();
```

Comment:
Perhaps a better message should be displayed here if it is a false positive. For example, that the registration has been blocked and you should contact an administrator.

Reply:
Good point, I've added a phrase that informs the user without giving away the offending value.